docs/guides/add-form.md

```diff
@@ -0,0 +1,304 @@
+    component={InputFieldFF}
+/>
+```
+
+### Switch field
+
+The switch field is a boolean. Provide a label, an initial value, and a help text. 
+
+```js
+<Field
+    type="checkbox"
+    name="newsletter"
+    label="I want to receive the newsletter"
+    component={SwitchFieldFF}
+    initialValue={false}
+/>
+```
+
+### Button
+
+Finally use the button component to submit the form:
+
+```js
+<Button type="submit" primary>
+    Submit form
+</Button>
+```
+
+## 4. Choose validators for the form component
+
+Validators are available via React Final Form. See [Validators](https://ui.dhis2.nu/demo/?path=/docs/forms-validators--page) in the UI library documentation for a list of validators.
+
+For our example, you'll validate that:
+- Surname, first name, email, and email confirmation are not empty
+- Email is a valid email address
+- Email confirmation matches the email address
+
+
+Remember to import your validators, as well:
+
+```js
+import {
+    composeValidators,
+    createEqualTo,
+    email,
+    hasValue,
+} from '@dhis2/ui'
+```
+
+### Validate non-empty fields
+
+We'll use `hasValue` to validate that our fields aren't empty. Add a `validate` prop to each field that requires this validation:
+
+```js
+<Field
+    required
+    name="surname"
+    label="Surname"
+    component={InputFieldFF}
+    validate={hasValue}
+/>
+
+<Field
+    required
+    name="firstname"
+    label="First name"
+    component={InputFieldFF}
+    validate={hasValue}
+/>
+
+<Field
+    required
+    name="email"
+    label="E-mail address"
+    component={InputFieldFF}
+    validate={hasValue}
+/>
+
+<Field
+    required
+    name="email-confirmation"
+    label="Confirm e-mail address"
+    component={InputFieldFF}
+    validate={hasValue}
+/>
+```
+
+### Validate an email address
+
+The `email` validator lets us confirm that an email address is valid. We'll add that validator to the email address field. Since we're now requiring multiple validations, we'll use `composeValidators()` to combine the validators into one.
+
+```js
+<Field
+    required
+    name="email"
+    label="E-mail address"
+    component={InputFieldFF}
+    validate={composeValidators(email, hasValue)}
+/>
+
+```
+
+### Validate matching email addresses
+
+We'll use the `createEqualTo` to validate that the email addresses match. We'll need to compose the validators for this field, as well.
+
+```js
+<Field
+    required
+    name="email-confirmation"
+    label="Confirm e-mail address"
+    component={InputFieldFF}
+    validate={composeValidators(
+        createEqualTo('email'),
+        hasValue
+    )}
+/>
+```
+
+## 5. Assemble the form
+
+The final form is assembled as follows:
+
+```js
+import {
+    Button,
+    InputFieldFF,
+    SingleSelectFieldFF,
+    SwitchFieldFF,
+    composeValidators,
+    createEqualTo,
+    email,
+    hasValue,
+} from '@dhis2/ui'
+import { ReactFinalForm } from '@dhis2/ui'
+
+<ReactFinalForm.Form onSubmit={alertValues}>
```

Comment:
This example looks really nice, but it won't work if someone copy-pastes it.  In particular, we should render this `jsx` in a functional React component and make sure that `alertValues` is defined so that this will compile and run.